### PR TITLE
pam/lattice regions

### PR DESCRIPTION
**Pull request by pford**
Issue #80 further region development with lattice regions.

Not complete yet, still need to update data when region/channel/stokes changes and have not tested regions outside the image region among other things.  Have only run backend ICD tests, have not tested interactively.

### CODE REVIEW

**Comment by pford** [comment]
Build failure: stage 3 upload package failed because no space left on device.

**Comment by pford** [comment]
Updated branch with protobuf changes

File: ImageData/HDF5Loader.h [hunk 1]
```diff
@@ -68,6 +69,11 @@ typename HDF5Loader::image_ref HDF5Loader::loadData(FileInfo::Data ds) {
     return dataSets[data];
 }
 
+bool HDF5Loader::getPixelMaskSlice(casacore::Array<bool>& mask, const casacore::Slicer& slicer) {
+    // HDF5Lattice is not a masked lattice
```
Comment (confluence) [inline ImageData/HDF5Loader.h]: Possibly another reason for us to use an `HDF5Image`?

Comment (pford) [reply ImageData/HDF5Loader.h]: Yes, there are many reasons to use HDF5Image and casacore Images in general.  Will do this as soon as I finish the last bit of lattice region implementation.

Comment (confluence) [reply ImageData/HDF5Loader.h]: OK. But please hold off until after the hdf5 stats branch has been merged -- it makes a lot of changes to the HDF5 loader code (although this does not affect the current type of the main HDF5 image object).

Comment (pford) [reply ImageData/HDF5Loader.h]: Sorry, saw your comment after merge was done.

Comment (confluence) [reply ImageData/HDF5Loader.h]: Don't worry; I meant hold off on making changes to the image type created by the HDF5 loader. This merge is fine.

**Pull request merged**